test-d/is-tuple.ts

```diff
@@ -0,0 +1,47 @@
+import {expectType} from 'tsd';
+import type {IsTuple} from '../index';
+
+// Tuples
+expectType<IsTuple<[]>>(true);
+expectType<IsTuple<[1]>>(true);
+expectType<IsTuple<[1, 2]>>(true);
+expectType<IsTuple<[1, 2, ...number[]]>>(true);
+expectType<IsTuple<[1?]>>(true);
+expectType<IsTuple<[1?, 2?]>>(true);
+expectType<IsTuple<[1?, 2?, ...number[]]>>(true);
+expectType<IsTuple<[never]>>(true);
```

Comment:
Actually some more:

```typescript
// Labeled tuples
expectType<IsTuple<[x: string, y: number]>>(true);
expectType<IsTuple<[first: string, ...rest: number[]]>>(true);
expectType<IsTuple<readonly [name: string, age?: number]>>(true);

// Mixed optional/required elements
expectType<IsTuple<[string, number?]>>(true);
expectType<IsTuple<[string?, number]>>(true);
expectType<IsTuple<readonly [string, number?, ...boolean[]]>>(true);

// Complex unions
expectType<IsTuple<[string] | [number] | number[]>>({} as boolean);
expectType<IsTuple<[string, ...number[]] | [number?, string?]>>(true);
expectType<IsTuple<[] | [number] | readonly [string]>>(true);
```

Reply:
Added in [797116f](https://github.com/sindresorhus/type-fest/pull/1024/commits/797116ffabe546fca28c099bb4252796887d956b). Had to remove the second test case in `Mixed optional/required elements`, because the tuple `[string?, number]` isn't valid (`A required element cannot follow an optional element`).

Reply:
Also, replaced all `1`s and `2`s with `number`s and `string`s, just to make the tests a bit more diverse.

Refer [fc451bf](https://github.com/sindresorhus/type-fest/pull/1024/commits/fc451bfe0dae2e3e131a695c9e469b2641626ad8).